sdk/core/Azure.Core.Extensions/src/Internal/ClientInformation.cs

```diff
@@ -43,6 +44,12 @@ public TClient GetClient(TOptions options, TokenCredential tokenCredential)
                     return _cachedClient;
                 }
 
+
+                if (RequiresTokenCredential && tokenCredential == null)
+                {
+                    throw new InvalidOperationException("Client registration requires a token credential. Configure it using UseCredential method.");
```

Comment:
Nit - probably useful to say `TokenCredential` instead of `token credential` to guide users.